hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/spi/impl/TestOzoneManagerServiceProviderImpl.java

```diff
@@ -411,18 +412,54 @@ public void testSyncDataFromOMDeltaUpdates() throws Exception {
         ArgumentCaptor.forClass(ReconTaskStatus.class);
     verify(reconTaskStatusDaoMock, times(1))
         .update(captor.capture());
-    assertTrue(captor.getValue().getTaskName().equals(OmDeltaRequest.name()));
+    assertEquals(OmDeltaRequest.name(), captor.getValue().getTaskName());
 
     verify(reconTaskControllerMock, times(1))
         .consumeOMEvents(any(OMUpdateEventBatch.class),
             any(OMMetadataManager.class));
     assertEquals(0, metrics.getNumSnapshotRequests().value());
   }
 
+  @Test
+  public void testSyncDataFromOMFullSnapshotForSNNFE() throws Exception {
+
+    // Non-Empty OM DB to start with.
+    ReconOMMetadataManager omMetadataManager = getTestReconOmMetadataManager(
+        initializeNewOmMetadataManager(temporaryFolder.newFolder()),
+        temporaryFolder.newFolder());
+    ReconTaskStatusDao reconTaskStatusDaoMock =
+        mock(ReconTaskStatusDao.class);
+    doNothing().when(reconTaskStatusDaoMock)
+        .update(any(ReconTaskStatus.class));
+
+    ReconTaskController reconTaskControllerMock = getMockTaskController();
+    when(reconTaskControllerMock.getReconTaskStatusDao())
+        .thenReturn(reconTaskStatusDaoMock);
+    doNothing().when(reconTaskControllerMock)
+        .reInitializeTasks(omMetadataManager);
+
+    OzoneManagerProtocol protocol = getMockOzoneManagerClientWithThrow();
+    OzoneManagerServiceProviderImpl ozoneManagerServiceProvider =
+        new MockOzoneServiceProvider(configuration, omMetadataManager,
+            reconTaskControllerMock, new ReconUtils(), protocol);
+
+    OzoneManagerSyncMetrics metrics = ozoneManagerServiceProvider.getMetrics();
+
+    // Should trigger full snapshot request.
+    ozoneManagerServiceProvider.syncDataFromOM();
+
+    ArgumentCaptor<ReconTaskStatus> captor =
+        ArgumentCaptor.forClass(ReconTaskStatus.class);
+    verify(reconTaskStatusDaoMock, times(1))
+        .update(captor.capture());
+    assertEquals(OmSnapshotRequest.name(), captor.getValue().getTaskName());
+    verify(reconTaskControllerMock, times(1))
+        .reInitializeTasks(omMetadataManager);
+    assertEquals(1, metrics.getNumSnapshotRequests().value());
+  }
```

Comment:
from this unit test, I learn a lot!! thanks Xu for writing this!

Reply:
> from this unit test, I learn a lot!! thanks Xu for writing this!

You are welcome. I just followed the style from the other unit tests LOL